nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -496,6 +500,33 @@ Costmap2DROS::updateMap()
   }
 }
 
+void
+Costmap2DROS::updateAndPublishMap()
```

Comment:
Please rebase onto to the latest `main` branch: there is a new indicator variable and behavior used to verificate that Costmap2DROS was started. It ought to be added to `updateAndPublishMap()`, since this API logic should be called only of Costmap2DROS was already started:
```
void
Costmap2DROS::updateAndPublishMap()
{
  if (stopped_) {
    // Do not execute is start() did not complete plugins activation
    return;
  }

  nav2_util::ExecutionTimer timer;

  // Measure the execution time of the updateMap method
  timer.start();
  updateMap();
  timer.end();

  ...
}

```